daemon/policy.go

```diff
@@ -244,6 +244,14 @@ func (d *Daemon) PolicyAdd(rules policyAPI.Rules, opts *AddOptions) (uint64, err
 		}
 	}
 	rev := d.policy.AddListLocked(rules)
+
+	// The rules are added, we can begin ToFQDN DNS polling for them
+	// Note: api.FQDNSelector.sanitize checks that the matchName entries are
+	// valid. This error should never happen (of course).
+	if err := d.dnsRuleGen.StartManageDNSName(rules); err != nil {
```

Comment:
Since we're moving this code anyway, I'd like to also make the `StartManageDNSName` happen before the `StopManageDNSName` (like the 1.2/1.3 fixes in https://github.com/cilium/cilium/pull/6997/files#diff-afacd8d4c41a906545f4ec8cde864b7fR268 and https://github.com/cilium/cilium/pull/6998/files#diff-afacd8d4c41a906545f4ec8cde864b7fR298). This should avoid the confusing message claiming that the DNS name is new every time we update. What do you think?